src/SourceBuild/content/eng/tools/BinaryToolKit/BinaryTool.CompareBinariesAgainstBaselines.cs

```diff
@@ -0,0 +1,130 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.FileSystemGlobbing;
+using Microsoft.Extensions.Logging;
+
+namespace BinaryToolKit;
+
+public partial class BinaryTool
```

Comment:
I initially used separate classes, but following [Matt's suggestion](https://github.com/dotnet/installer/pull/18726#discussion_r1502793817), I decided to discard the independent classes in favor of a single stateless class, with the only exception being the logger. I think that the tasks (Detect, Compare, Remove) need be separated in some way, but I have reservations about using separate classes, considering that BinaryTool.ExecuteAsync() is the sole method that invokes the methods for the tasks. Because of this, I prefer keeping the methods for the tasks private, for instance, CompareBinariesAgainstBaselines.cs is private. If I were to use separate classes, I could create internal classes, but this would make the task methods accessible to the CLI.